phoenix-core/src/it/java/org/apache/phoenix/end2end/OrphanViewToolIT.java

```diff
@@ -0,0 +1,456 @@
+            ResultSet rs = connection.createStatement().executeQuery(viewQuery);
+            assertTrue(rs.next());
+            assertTrue(rs.getLong(1) == fanout + fanout * fanout + fanout * fanout * fanout);
+            deleteAllRows(connection, SCHEMA1, SCHEMA2, SCHEMA3, SCHEMA4);
+        }
+    }
+
+    private void verifyNoChildLink(Connection connection, String viewSchemaName) throws Exception {
+        // Verify that there there is no link in the system child link table
+        String childLinkQuery = "SELECT COUNT(*) FROM " + SYSTEM_CHILD_LINK_NAME +
+                " WHERE " + TABLE_SCHEM + " = '" + viewSchemaName + "' AND " +
+                LINK_TYPE + " = " + PTable.LinkType.CHILD_TABLE.getSerializedValue();
+        ResultSet rs = connection.createStatement().executeQuery(childLinkQuery);
+        assertTrue(rs.next());
+        assertTrue(rs.getLong(1) == 0);
+    }
+
+    private void verifyNoViewNoLink(Connection connection, String viewSchemaName) throws Exception {
+        // Verify that the views and links have been removed from the system catalog table
+        String catalogQuery = "SELECT COUNT(*) FROM " + SYSTEM_CATALOG_NAME +
+                " WHERE " + TABLE_SCHEM + " = '" + viewSchemaName + "'";
+        ResultSet rs = connection.createStatement().executeQuery(catalogQuery);
+        assertTrue(rs.next());
+        long count = rs.getLong(1);
+        assertTrue(rs.getLong(1) == 0);
+    }
+
+    @Test
+    public void testDeleteBaseTableRows() throws Exception {
+        String baseTableName = generateUniqueName();
+        String baseTableFullName = SchemaUtil.getTableName(SCHEMA1, baseTableName);
+        try (Connection connection = DriverManager.getConnection(getUrl());
+             Connection viewConnection =
+                     isMultiTenant ? DriverManager.getConnection(TENANT_SPECIFIC_URL) : connection) {
+            createBaseTableAndViews(connection, baseTableFullName, viewConnection, SCHEMA2, SCHEMA2, SCHEMA2);
+            // Delete the base table rows from the system catalog
+            String deleteTableRows = "DELETE FROM " + SYSTEM_CATALOG_NAME +
+                    " WHERE " + TABLE_SCHEM + " = '" + SCHEMA1 + "' AND " +
+                    TABLE_NAME + " = '" + baseTableName + "' AND " +
+                    TABLE_TYPE + " = '" + PTableType.TABLE.getSerializedValue() + "'";
+            connection.createStatement().execute(deleteTableRows);
+            connection.commit();
+            // Verify that the views we have created are still in the system catalog table
+            String viewQuery = "SELECT COUNT(*) FROM " + SYSTEM_CATALOG_NAME +
+                    " WHERE " + TABLE_TYPE + " = '" + PTableType.VIEW.getSerializedValue() + "'";
+            ResultSet rs = connection.createStatement().executeQuery(viewQuery);
+            assertTrue(rs.next());
+            long count = rs.getLong(1);
```

Comment:
It looks like `count` is unused. Please remove.

Reply:
I will remove it